compiler/backend/src/org/jetbrains/kotlin/codegen/ExpressionCodegen.java

```diff
@@ -3833,6 +3833,9 @@ public Unit invoke(InstructionAdapter v) {
                                 v.mark(nonnull);
                             }
                         }
+                        else if (value.type == Type.VOID_TYPE) {
+                            v.aconst(null);
+                        }
                         else {
                             v.dup();
                             generateInstanceOfInstruction(rightType);
```

Comment:
Similar code is present in `generateInstanceOf` which probably means `Unit is Int?` is also affected?